packages/react-component-library/src/components/DataGrid/DataGrid.tsx

```diff
@@ -56,12 +56,14 @@ export interface DataGridBaseProps<T extends object>
   initialRowSelection?: RowSelectionState
   isFullWidth?: boolean
   hasHover?: boolean
+  hasRowsPerPage?: boolean
```

Comment:
There's an issue with the interface here because if `hasRowsPerPage` is `true` then `pageSize` is redundant. Normally we would solve this through union types but there is a fair amount of complexity already with `DataGrid` types. Another option could be that we always display the `RowsPerPage` and mark to `pageSize` prop as deprecated.

Reply:
Can we check it plays nice with the external state pagination stories?